test/helpers/replicate.js

```diff
@@ -7,33 +7,73 @@ const pull = require('pull-stream')
 const pullMany = require('pull-many')
 const deepEqual = require('fast-deep-equal')
 
+module.exports = async function replicate(...peers) {
+  if (peers.length === 1 && Array.isArray(peers[0])) peers = peers[0]
+  if (peers.length === 2) return replicatePair(...peers)
+
+  return pull(
+    pull.values(peers),
+    pull.asyncMap((person1, cb) => {
+      pull(
+        pull.values(peers),
+        pull.asyncMap((person2, cb) => {
+          if (person1.id === person2.id) return cb(null, true)
+
+          replicatePair(person1, person2)
+            .then((res) => cb(null, true))
+            .catch((err) => cb(err))
+        }),
+        pull.collect(cb)
+      )
+    }),
+    pull.collectAsPromise()
+  )
+}
+
 /**
  * Fully replicates person1's metafeed tree to person2 and vice versa
  */
-module.exports = async function replicate(person1, person2) {
+async function replicatePair(person1, person2) {
+  // const start = Date.now()
+  // const ID = [person1, person2]
+  //   .map(p => p.name || p.id.slice(0, 10))
+  //   .join('-')
+
   // Establish a network connection
   const conn = await p(person1.connect)(person2.getAddress())
 
+  const isSync = await ebtReplicate(person1, person2).catch((err) =>
+    console.error('Error with ebtReplicate:\n', err)
+  )
+  if (!isSync) {
+    console.error('EBT failed to replicate! Final state:')
+    console.log(person1.id, await p(person1.getVectorClock)())
+    console.log(person2.id, await p(person2.getVectorClock)())
+  }
+
+  await p(conn.close)(true).catch(console.error)
+  // const time = Date.now() - start
+  // const length = Math.max(Math.round(time / 100), 1)
+  // console.log(ID, Array(length).fill('▨').join(''), time + 'ms')
+}
+
+async function ebtReplicate(person1, person2) {
   // ensure persons are replicating all the trees in their forests,
   // from top to bottom
   const stream = setupFeedRequests(person1, person2)
 
   // Wait until both have replicated all feeds in full (are in sync)
-  let i = 0
-  await retryUntil(async () => {
-    i++
+  const isSync = async () => {
     const clocks = await Promise.all([
       p(person1.getVectorClock)(),
       p(person2.getVectorClock)(),
     ])
```

Comment:
This replicate function will probably have the same bug that we ran into in https://github.com/ssbc/ssb-tribes2/pull/106#discussion_r1196639404

I think this might break if it's bob that creates the group.